clients/src/main/java/org/apache/kafka/common/network/SslTransportLayer.java

```diff
@@ -500,8 +500,11 @@ public int read(ByteBuffer dst) throws IOException {
             read = readFromAppBuffer(dst);
         }
 
-        int netread = 0;
-        if (dst.remaining() > 0) {
+        boolean isClosed = false;
+        // Keep reading from the socket into the dst buffer until either the dst buffer is full or there is no more
+        // data in the socket. Each loop read at most once from the socket.
+        while (dst.remaining() > 0) {
```

Comment:
I think we should document the behaviour in the javadoc of `read`.

Reply:
Previously I thought this maybe an implementation detail which does not need to stay in the Java doc. Sure. I have updated the patch to move the comment to Java doc. Thanks for the suggestion.

Reply:
Thanks. Since this affects consumer performance and we added tests to verify the behaviour, I see it as part of the specification now.

Reply:
I see. Good point.